File: Day4/C/Ques16.c
#include <stdio.h>

int main()
{
    int l, u;

    printf("Enter range (lower upper): ");
    scanf("%d %d", &l, &u);

    for(int n = l; n <= u; n++)
    {
        int count = 0, sum = 0;
        int a = n, b = n;

        int temp_num = n;
        while(temp_num > 0)
        {
            count++;
            temp_num /= 10;
        }

        while(a > 0)
        {
            int digit = a % 10;

            int power = 1;
            for(int i = 0; i < count; i++)
            {
                power *= digit;
            }

            sum += power;
            a /= 10;
        }

        if(sum == b)
        {
            printf("%d ", b);
        }
    }

    return 0;
}

File: Day4/Python/Ques16.py
import math
l=int(input("Enter positive lower range:"))
u=int(input("Enter positive upper range:"))
print("Armstrong Number in range",l,"-","u")
for i in range (l,u+1):
    a=b=temp=i
    count=s=0
    while temp>0 :
        temp=temp//10
        count += 1
    while a>0:
        s=s+ math.pow(a%10,count)
        a=a//10
    if s==b:
        print(i,end=" ")


